### PR TITLE
feat: image type and upload component

### PR DESCRIPTION
**Pull request by makeitrein**
Demo Video: https://share.cleanshot.com/bHlxMcmk

### CODE REVIEW

File: packages/action-schema/src/action.ts [hunk 1]
```diff
@@ -37,9 +37,15 @@ interface StringValue {
   value: string
 }
 
+interface ImageValue {
```
Comment (makeitrein) [inline packages/action-schema/src/action.ts:40]: The files below are mainly ceremony to add a new value type

File: apps/web/modules/components/entity/edit-events.ts [hunk 2]
```diff
@@ -449,6 +494,32 @@ const listener =
...
+        const newValue = { ...triple.value, value: imageSrc };
+
+        return update(
+          Triple.ensureStableId({
+            ...triple,
+            value: newValue,
+          }),
+          triple
+        );
+      }
```
Comment (baiirun) [inline apps/web/modules/components/entity/edit-events.ts:522]: Is the difference between `CREATE_IMAGE_TRIPLE_WITH_VALUE` and `UPLOAD_IMAGE` basically the difference between placeholder and non-placeholder image fields?

File: apps/web/modules/components/entity/editable-fields.tsx [hunk 3]
```diff
@@ -81,3 +84,142 @@ export function PageStringField({ ...props }: PageStringFieldProps) {
...
+    },
+    avatar: {
+      height: 44,
+      width: 44,
+    },
+    banner: {
+      height: 44,
+      width: 240,
+    },
+  };
```
Comment (baiirun) [inline apps/web/modules/components/entity/editable-fields.tsx:108]: These styles can probably live outside the component so we're not recreated them every time the component re-renders.

File: apps/web/modules/components/entity/edit-events.ts (hunk 2 above)
Comment (makeitrein) [reply apps/web/modules/components/entity/edit-events.ts:522]: Less placeholders, more like in the editable entity page, we have an existing triple to upload to, but in the table, it's everything (triple + value) must be created at once

File: apps/web/modules/components/entity/editable-fields.tsx (hunk 3 above)
Comment (makeitrein) [reply apps/web/modules/components/entity/editable-fields.tsx:108]: Yup yup

File: apps/web/modules/components/entity/edit-events.ts [hunk 4]
```diff
@@ -247,15 +261,20 @@ const listener =
       case 'CHANGE_TRIPLE_TYPE': {
         const { type, triples } = event.payload;
 
+        const value = emptyValue(type);
```
[the remark this answers is not in the document]
Comment (makeitrein) [reply apps/web/modules/components/entity/edit-events.ts:269]: Good catch

File: apps/web/modules/components/entity/entity-page-utils.tsx [hunk 5]
```diff
@@ -9,30 +9,30 @@ export function sortEntityPageTriples(visibleTriples: Triple[], schemaTriples: T
 
   /* Visible triples includes both real triples and placeholder triples */
   return visibleTriples.sort((tripleA, tripleB) => {
```
[the remark this answers is not in the document]
Comment (makeitrein) [reply apps/web/modules/components/entity/entity-page-utils.tsx:11]: Added some clarity in the comment above

[the remark this answers is not in the document]
Comment (baiirun) [reply apps/web/modules/components/entity/entity-page-utils.tsx:11]: It's still not clear to me what the actual sort order is without diving through the code

**Pull request merged**